PoESkillTree.Engine.GameModel.Tests/PassiveTree/Base/JsonPassiveNodeTest.cs

```diff
@@ -0,0 +1,191 @@
+﻿using NUnit.Framework;
+using System;
+
+namespace PoESkillTree.Engine.GameModel.PassiveTree.Base
+{
+    public class JsonPassiveNodeTest
+    {
+        [TestCase(CharacterClass.Scion)]
+        [TestCase(CharacterClass.Marauder)]
+        [TestCase(CharacterClass.Ranger)]
+        [TestCase(CharacterClass.Witch)]
+        [TestCase(CharacterClass.Duelist)]
+        [TestCase(CharacterClass.Templar)]
+        [TestCase(CharacterClass.Shadow)]
+        [TestCase(null)]
+        public void JsonPassiveNode_StartingCharacterClass(CharacterClass? characterClass)
+        {
+            var node = new JsonPassiveNode()
+            {
+                StartingCharacterClass = characterClass
+            };
+
+            Assert.AreEqual(characterClass.HasValue, node.IsRootNode);
+            Assert.AreEqual(characterClass, node.StartingCharacterClass);
+        }
+
+        [TestCase("", ExpectedResult = false)]
+        [TestCase("testing", ExpectedResult = true)]
+        public bool JsonPassiveNode_IsAscendancyNode(string ascendancyName)
+        {
+            var node = new JsonPassiveNode
+            {
+                AscendancyName = ascendancyName
+            };
+
+            return node.IsAscendancyNode;
+        }
+
+        [TestCase(PassiveNodeType.Keystone, ExpectedResult = PassiveNodeType.Keystone)]
+        [TestCase(PassiveNodeType.Mastery, ExpectedResult = PassiveNodeType.Mastery)]
+        [TestCase(PassiveNodeType.Notable, ExpectedResult = PassiveNodeType.Notable)]
+        [TestCase(PassiveNodeType.JewelSocket, ExpectedResult = PassiveNodeType.JewelSocket)]
+        [TestCase(PassiveNodeType.Small, ExpectedResult = PassiveNodeType.Small)]
+        [TestCase(PassiveNodeType.Keystone | PassiveNodeType.Mastery | PassiveNodeType.Notable | PassiveNodeType.JewelSocket | PassiveNodeType.Small, ExpectedResult = PassiveNodeType.Small)]
+        [TestCase(42, ExpectedResult = PassiveNodeType.Small)]
+        public PassiveNodeType JsonPassiveNode_PassiveNodeType(PassiveNodeType passiveNodeType)
+        {
+            var node = new JsonPassiveNode
+            {
+                IsKeystone = passiveNodeType == PassiveNodeType.Keystone,
+                IsMastery = passiveNodeType == PassiveNodeType.Mastery,
+                IsNotable = passiveNodeType == PassiveNodeType.Notable,
+                IsJewelSocket = passiveNodeType == PassiveNodeType.JewelSocket,
+            };
+
+            return node.PassiveNodeType;
+        }
+
+        [TestCase(0, new[] { 1f })]
+        [TestCase(0, new[] { 1f })]
+        [TestCase(0, new[] { 1f })]
+        [TestCase(0, new[] { 1f })]
+        [TestCase(1, new[] { 1f, 2f })]
+        [TestCase(1, new[] { 1f, 2f })]
+        [TestCase(1, new[] { 1f, 2f })]
+        [TestCase(1, new[] { 1f, 2f })]
```

Comment:
These 8 test cases are only 2 distinct ones.

Reply:
Not sure what happened here...

Reply:
This is fixed.